core/src/main/java/org/keycloak/representations/JsonWebToken.java

```diff
@@ -161,6 +162,17 @@ public JsonWebToken audience(String... audience) {
         return this;
     }
 
+    public JsonWebToken addAudience(String audience) {
+        if (this.audience == null) {
+            this.audience = new String[] { audience };
```

Comment:
Yes, Set is better. 

But it will require to change signature of JsonWebToken.getAudience/setAudience methods, so kept the String[] in the signatures for backwards compatibility (people may already use JsonWebToken directly in their application). Also the "aud" can be single String or an Array per JSON and hence we use this custom "StringOrArraySerializer" / "StringOrArrayDeserializer", which would probably need to be rewritten or different serializer used etc.

So the easiest would be to handle duplicates in the "addAudience" method. If you prefer to change it to Set, I can take a look at it (will require some work with the Serializer. For backwards compatibility it is probably ok if we just document in the migration guide). WDYT?

Reply:
I think that handle duplicates is enough. Didn't consider backward compatibility, good point.

Reply:
@pedroigor I've updated PR to handle duplicates.